bpf/estat/metaslab-alloc.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 Delphix. All rights reserved.
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include <sys/metaslab.h>
+#include <sys/vdev_impl.h>
+#include <sys/spa_impl.h>
+
+#define VD_NAME_SIZE 32
+typedef struct {
+	u64 ts;
+	u64 size;
+	u64 asize;
+	u64 alloc_time;
+	char vd_name[VD_NAME_SIZE];
+} data_t;
+
+BPF_HASH(data_map, u32, data_t);
+
+#ifndef OPTARG
+#define	POOL "domain0"
+#else
+#define	POOL (OPTARG)
+#endif
+
+static inline bool
+equal_to_pool(char *str)
+{
+	char comparand[sizeof (POOL)];
+	bpf_probe_read(&comparand, sizeof (comparand), str);
+	char compare[] = POOL;
+
+	for (int i = 0; i < sizeof (comparand); ++i)
+		if (compare[i] != comparand[i])
+			return (false);
+
+	return (true);
+}
+
+// @@ kprobe|metaslab_alloc_dva|metaslab_alloc_dva_entry
+int
+metaslab_alloc_dva_entry(struct pt_regs *ctx,
+	spa_t *spa, metaslab_class_t *mc, uint64_t psize)
```

Comment:
[super nit] If we're approximating illumos cstyle, the continuation line should be indented 4 spaces.